File: week2/p9.py
def is_identity_matrix(mat):
    n = len(mat)
    for i in range(n):
        for j in range(n):
            if i == j and mat[i][j] != 1:  # diagonal element check
                return False
            elif i != j and mat[i][j] != 0:  # non-diagonal check
                return False
    return True

# Example usage
n = int(input())
mat = [list(map(int, input().split())) for _ in range(n)]
print(is_identity_matrix(mat))

File: week2/p9.c
#include <stdio.h>

int isIdentityMatrix(int mat[][100], int n) {
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            if (i == j && mat[i][j] != 1) // diagonal element should be 1
                return 0;
            else if (i != j && mat[i][j] != 0) // non-diagonal should be 0
                return 0;
        }
    }
    return 1;
}

int main() {
    int n;
    scanf("%d", &n);
    int mat[100][100];
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            scanf("%d", &mat[i][j]);

    if (isIdentityMatrix(mat, n))
        printf("True\n");
    else
        printf("False\n");

    return 0;
}
